simpleclient_pushgateway/src/main/java/io/prometheus/client/exporter/PushGateway.java

```diff
@@ -54,22 +50,53 @@
  */
 public class PushGateway {
 
-  private final String address;
   private static final int SECONDS_PER_MILLISECOND = 1000;
+
+  private final String address;
+
+  private final String port;
+
+  private final String username;
+
+  private final String password;
+
+  private final String scheme;
+
+  /**
+   * Construct a Pushgateway, with the given address.
+   * <p>
+   *
+   * @param address host or ip of the Pushgateway.
+   * @param port    port of the Pushgateway.
+   */
+  public PushGateway(String address, String port) {
+    this("http", address, port, null, null);
+  }
+
   /**
    * Construct a Pushgateway, with the given address.
    * <p>
-   * @param address  host:port or ip:port of the Pushgateway.
+   *
+   * @param scheme   url scheme of the Pushgateway.
+   * @param address  host or ip of the Pushgateway.
+   * @param port     port of the Pushgateway.
+   * @param username username for basic authorization.
+   * @param password password for basic authorization.
    */
-  public PushGateway(String address) {
+  public PushGateway(String scheme, String address, String port, String username, String password) {
```

Comment:
You're presuming that basic auth is the only way that a use would like to authenticate. Someone might want ssl, kerberos or something else. Can you do something more generic?